libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -54,15 +56,59 @@ internal class MapboxRerouteController(
             observers.forEach { it.onRerouteStateChanged(field) }
         }
 
+    private companion object {
+        private const val TAG = "MbxRerouteController"
+
+        /**
+         * Max dangerous maneuvers radius meters. See [RouteOptions.avoidManeuverRadius]
+         */
+        private const val MAX_DANGEROUS_MANEUVERS_RADIUS = 1000
+
+        /**
+         * Apply reroute options. Speed must be provided as **m/s**
+         */
+        private fun RouteOptions?.applyRerouteOptions(
+            rerouteOptions: RerouteOptions,
+            speed: Float?
+        ): RouteOptions? {
+            if (this == null || speed == null) {
```

Comment:
nit: make `this` non-null.

Reply:
it's a bit tricky: I didi it to avoid do smt like `routeOptions?.applyBlaBla() ?: routeOptions`

Reply:
But you still have to do this, because `null` is returned in this case.